AutomatedTesting/cmake/VersionUtils.cmake

```diff
@@ -0,0 +1,116 @@
+# {BEGIN_LICENSE}
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+# {END_LICENSE}
+# This file is copied during engine registration. Edits to this file will be lost next
+# time a registration happens.
+
+include_guard()
+
+function(o3de_version_specifier_contains op specifier_version version result)
+    set(contains_version FALSE)
+
+    if(op STREQUAL "==" AND version VERSION_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "!=" AND NOT version VERSION_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "<=" AND version VERSION_LESS_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL ">=" AND version VERSION_GREATER_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "<" AND version VERSION_LESS specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL ">" AND version VERSION_GREATER specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "===" AND version VERSION_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "~=")
+        # compatible versions have an equivalent combination of >= and == 
+        # e.g. ~=2.2 is equivalent to >=2.2,==2.*
```

Comment:
So this is any value between `2.*` and `3.0` according to the [compatible release](https://peps.python.org/pep-0440/#compatible-release) clause for PEP440?

Reply:
>= 2.2 ,== 2.*  means from 2.2 (inclusive) to 3 (exclusive).  This example from the docs:
```
For example, the following groups of version clauses are equivalent:

~= 2.2
>= 2.2, == 2.*
```
... informed the code based on the [packaging implementation here](https://github.com/pypa/packaging/blob/1b9da4420e9e23046dc095a3779f2607ae6e4d62/src/packaging/specifiers.py#L376-L395) and the [Dependency implementation here](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Dependency/Dependency.inl#L166-L188).

Reply:
So a version of `2.1` would fail the check?

Reply:
yup, `2.1` would fail and also `3.0`